pkg/config/config.go

```diff
@@ -0,0 +1,119 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/spf13/viper"
+)
+
+const configName = "cli_config"
+const defaultConfigFileType = "json"
+const appData = "AppData"
+
+// var configPath string
+
+func Setup() {
+	viper.SetDefault(constants.ConfigHost, "")
+	viper.SetDefault(constants.ConfigApiKey, "")
+	viper.SetDefault(constants.ConfigSpace, "")
+	viper.SetDefault(constants.ConfigNoPrompt, false)
+	//	viper.SetDefault(constants.ConfigProxyUrl, "")
+	//	viper.SetDefault(constants.ConfigShowOctopus, true)
+	viper.SetDefault(constants.ConfigOutputFormat, "table")
+
+	viper.SetConfigName(configName)
+	viper.SetConfigType(defaultConfigFileType)
+
+	if runtime.GOOS == "windows" {
+		viper.SetDefault(constants.ConfigEditor, "notepad")
+	} else { // unix
+		viper.SetDefault(constants.ConfigEditor, "nano")
+	}
+	getConfigPath()
```

Comment:
useless call to `getConfigPath()`

Reply:
this is used to set the config path in viper. Will make this more clear.